cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/builder/Fluent.kt

```diff
@@ -222,6 +362,72 @@ fun LanguageFrontend.call(
     return node
 }
 
+/**
+ * Creates a new [CallExpression] (or [MemberCallExpression]) in the Fluent Node DSL with the given
+ * [name] and adds it to the nearest enclosing [Holder]. Depending on whether it is a
+ * [StatementHolder] it is added to the list of [StatementHolder.statements] or in case of an
+ * [ArgumentHolder], the function [ArgumentHolder.addArgument] is invoked.
+ *
+ * The type of expression is determined whether [name] is either a [Name] with a [Name.parent] or if
+ * it can be parsed as a FQN in the given language. It also automatically creates either a
+ * [DeclaredReferenceExpression] or [MemberExpression] and sets it as the [CallExpression.callee].
+ * The [init] block can be used to create further sub-nodes as well as configuring the created node
+ * itself.
+ */
+context(Holder<out Statement>)
+
+fun LanguageFrontend.construct(
+    name: CharSequence,
+    init: (ConstructExpression.() -> Unit)? = null
+): ConstructExpression {
+    val node = newConstructExpression(parseName(name))
+    node.type = t(name)
+
+    if (init != null) {
+        init(node)
+    }
+
+    val holder = this@Holder
+    if (holder is StatementHolder) {
+        holder += node
+    } else if (holder is ArgumentHolder) {
+        holder += node
+    }
+
+    return node
+}
+
+context(Holder<out Statement>)
+
+fun LanguageFrontend.new(init: (NewExpression.() -> Unit)? = null): NewExpression {
+    val node = newNewExpression()
+    if (init != null) init(node)
+
+    val holder = this@Holder
+    if (holder is StatementHolder) {
+        holder += node
+    } else if (holder is ArgumentHolder) {
+        holder += node
+    }
+    return node
+}
+
+fun LanguageFrontend.memberOrRef(
+    name: Name,
+    type: Type = UnknownType.getUnknownType()
+): Expression {
+    val node =
+        if (name.parent != null) {
+            newMemberExpression(name.localName, memberOrRef(name.parent))
+        } else {
+            newDeclaredReferenceExpression(name.localName, parseType(name.localName))
+        }
+    node.type = type
+
+    node.isStaticAccess = name.localName.first().isUpperCase()
```

Comment:
What's the logic behind that?

Reply:
`isStaticAccess` is somehow used to avoid resolving a variable for a `DeclaredReferenceExpression`. E.g., the `VariableUsageResolver` adds a field `System` to the enclosing TR if there's a call `System.out.println` (or similar). Setting this variable avoids it and won't resolve the symbol. No idea if this is always intended and if there's a better way to do this but I would be more than happy to have an alternative (and clearly documented) solution

Reply:
Yes but in this case you should just set isStaticAccess in the node's init function (for `System...`) instead of relying on this weird logic here; because for example in Go upper and lowercase variable names do have semantics, but definitely not static/non-static.